### PR TITLE
Implement map wrangke using custom wrangle agent

### PR DESCRIPTION
**Pull request by mborodii-prog**
### 
Add map wrangle with tests and notebook guide

**Summary**

Introduces a new map wrangle for semantic column name mapping in recipes.
Adds comprehensive recipe-level tests and a user-facing guidance notebook.
Handles edge cases (identity mappings, conflicting targets) and supports case sensitivity.

**Motivation**

Simplifies aligning messy or varied source column names to a standard schema.
Reduces manual renaming and improves recipe portability across datasets.

**Changes**

New wrangle: [main.py](https://orange-potato-7vjqv5w6qjw4fxq4q.github.dev/) (map).
Tests: [test_main.py](https://orange-potato-7vjqv5w6qjw4fxq4q.github.dev/) (TestMap).
Docs: [map-wrangle-guide.ipynb](https://orange-potato-7vjqv5w6qjw4fxq4q.github.dev/).
Implementation

Greedy unique matching between input columns and targets.
Skips identity self-mappings to avoid reserving targets unnecessarily.
Drops existing conflicting target columns before renaming to prevent duplicates.
Similarity blends SequenceMatcher ratio with target-token coverage; no stopwords.

**Parameters:**
targets: list[str], required.
input: optional list; supports “?” suffix for optional input.
threshold: float [0–1], defaults 0.6.
drop_unmapped: bool.
case_sensitive: bool, default False.
Tests

**Scenarios:** basic mapping, subset input, threshold filtering, conflict resolution, overwriting existing targets, invalid parameters, case-insensitive and case-sensitive behavior.

**Documentation**

Notebook demonstrates usage end-to-end with assertions and includes sections for error handling and best-practice tips.
Usage Example
```
wrangles:
  - map:
      targets:
        - name
        - price
        - description
      input:
        - Product Name
        - Unit Price?
        - Item Desc
      threshold: 0.6
      drop_unmapped: false
      case_sensitive: false
```

Risks

Heuristic mapping may need tuning (threshold, case_sensitive) for certain datasets.
Large schemas incur more comparisons; mitigate by providing an input subset.

**Checklist**

- [x] Tests added and passing for map.
- [x] Backwards compatible (adds new functionality only).
- [x] Optional follow-up: add recipe schema entry and docstrings for map for discoverability.

### CODE REVIEW

File: wrangles/recipe_wrangles/main.py [hunk 1]
```diff
@@ -1549,6 +1550,168 @@ def resolve_input(candidate_list):
...
+    additionalProperties: false
+    required:
+      - targets
+    properties:
+      targets:
+        type: array
+        description: List of desired target column names to map to. Each target will be assigned at most once.
+        minItems: 1
+      items:
+        type: string
```
[the remark this answers is not in the document]
Comment (thomasstvr) [reply wrangles/recipe_wrangles/main.py:1576]: I don't get how items would fit under targets. Perhaps it shouldn't be there at all?

**Review by thomasstvr — CHANGES_REQUESTED** [review]
@mborodii-prog copilot made some good suggestions. I did not go through all 15 of them, but I'm sure some are better than others while some might be meaningless. Please go through copilot's comments and make changes as needed.

One thing that jumps out to me is that all targets should be created, regardless of meeting the threshold in order to match the functionality of a map wrangle. So, if nothing meets the threshold or there are more targets than existing columns, the wrangle should simply create those as empty columns.

This also seems like a good opportunity to use recipes within our code. I'm not sure how exactly, but it just seems like a recipe could be used for renaming matches and creating columns.

**Comment by ebhills** [comment]
Queue triage (2026-07-27)

- **Disposition:** Draft — delivery-owner action
- **Delivery owner:** @mborodii-prog
- **Next action:** Return to the accepted scope in issue #320. Resolve the merge conflict and 15 review threads, or close this PR and rebuild the feature as a focused change.

GitHub is the status record; update this PR rather than the external spreadsheet.
